sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchClientBuilder.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.batch;
+
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.BlobServiceClient;
+
+/**
+ *
+ */
+public final class BlobBatchClientBuilder {
+    private final ClientLogger logger = new ClientLogger(BlobBatchClientBuilder.class);
+
+    private String accountUrl;
+    private HttpPipeline pipeline;
+
+    /**
+     *
+     * @param client
+     */
+    public BlobBatchClientBuilder(BlobServiceClient client) {
+        this.accountUrl = client.getAccountUrl();
+        this.pipeline = client.getHttpPipeline();
+    }
+
+    /**
+     *
+     * @param client
+     */
+    public BlobBatchClientBuilder(BlobServiceAsyncClient client) {
```

Comment:
Shouldn't the builder be instantiated with a default constructor and the client should be a setter?

Reply:
This is following what `LeaseClientBuilder` currently does, we can switch this up a little in preview 5 if need be